### PR TITLE
Feature/emission deforestation

### PR DESCRIPTION
**Pull request by pjosh**
## Overview

The emissions widget should be a line chart, with time in years as the x-axis, and emissions (in tons of carbon, or tones of CO2) as the y-axis, on hover it should show the year and emissions value.

There should be a dynamic sentence which comments on the yearly average emissions over the selected year range.

## Demo

captura de pantalla 2018-02-02 a las 12 19 26

### CODE REVIEW

File: app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { createElement, PureComponent } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import isEqual from 'lodash/isEqual';
+import COLORS from 'pages/country/data/colors.json';
```
Comment (edbrett) [inline app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation.js]: If passing in colors from parent this isn't needed.

File: app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import WidgetEmissionsDeforestationComponent from './widget-emissions-deforestation-component';
+
+const mapStateToProps = ({ widgetEmissionsDeforestation }, ownProps) => {
+  const { settings, data } = widgetEmissionsDeforestation;
+  const { activeIndicator } = ownProps;
+  const selectorData = {
+    data: data.loss,
+    settings,
+    indicator: activeIndicator,
+    colors: COLORS.emissions
```
Comment (edbrett) [inline app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation.js]: Again here

File: app/javascript/utils/calculations.js [hunk 3]
```diff
@@ -2,6 +2,8 @@ export const tonsToTonnes = value => value * 0.907185;
 
 export const biomassToCO2 = value => (value + 0.489 * value ** 0.89) * 0.47;
 
+export const biomassToC = value => value + 0.489 * value ** 0.89;
+
```
Comment (edbrett) [inline app/javascript/utils/calculations.js]: Clap clap

File: app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation-selectors.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    const { startYear, endYear, metric, threshold } = settings;
+    const totalEmissions = data.map(d => d[metric]).reduce((sum, d) => sum + d);
+    const emissionType = metric === 'biomassCarbon' ? 'carbon' : 'CO2';
+    let indicatorText = '';
+    if (indicator.value === 'mining') {
+      indicatorText = ` in ${indicator.label.toLowerCase()} regions`;
+    } else if (indicator.value !== 'gadm28') {
+      indicatorText = ` in ${indicator.label.toLowerCase()}`;
+    }
+    return `Between <b>${startYear}</b> and <b>${endYear}</b>, <b>${totalEmissions /
```
Comment (edbrett) [inline app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation-selectors.js]: Please check formatting if this value.

File: app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation.js (hunk 1 above)
Comment (pjosh) [reply app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation.js]: oh you're right I forgot your changes

File: app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation-selectors.js (hunk 4 above)
Comment (pjosh) [reply app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation-selectors.js]: Is the same formatting as the notebook. Maybe the notebook is wrong?

Comment (edbrett) [reply app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation-selectors.js]: Yes most likely. We should be consistent with other widgets

Comment (pjosh) [reply app/javascript/pages/country/widget/widgets/widget-emissions-deforestation/widget-emissions-deforestation-selectors.js]: I agree!

**Comment by pjosh** [comment]
hi @benlaken ! I have the same values as the notebook, can you check it? Maybe something is wrong in the notebook and therefore also in the widget.

**Comment by benlaken** [comment]
You are right @pjosh  - in that case let's go with it, as we are correct in relation to the values in Charlie's table.

**Pull request merged**